streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -119,7 +121,7 @@
      *
      */
     public enum State implements ThreadStateTransitionValidator {
-        CREATED(1, 4, 5), RUNNING(2, 4, 5), PARTITIONS_REVOKED(2, 3, 4, 5), ASSIGNING_PARTITIONS(1, 4, 5), PENDING_SHUTDOWN(5), DEAD;
+        CREATED(1, 5), RUNNING(2, 5), PARTITIONS_REVOKED(2, 3, 5), ASSIGNING_PARTITIONS(4, 5), PARTITIONS_ASSIGNED(1, 2, 5), PENDING_SHUTDOWN(6), DEAD;
```

Comment:
Can we merge ASSIGNING_PARTITIONS to PARTITIONS_ASSIGNED? And if yes we'd document it in the upgrade guide doc as well.